src/gpu/nvidia/sycl_cuda_scoped_context.cpp

```diff
@@ -51,7 +52,10 @@ cuda_sycl_scoped_context_handler_t::
     // we need to release the placed_context_ since we set it from
     // ctx.get() retains the underlying context so we need to remove it
     try {
-        if (need_to_recover_) { CUDA_EXECUTE_FUNC(cuCtxSetCurrent, original_); }
+        if (need_to_recover_) {
+            CUDA_EXECUTE_FUNC(cuDevicePrimaryCtxRelease, currentDevice);
```

Comment:
It looks like there is one more place where we query CUcontext:
https://github.com/oneapi-src/oneDNN/blob/18932605309f43966bf7b01ce4b279cb2ac77c68/src/gpu/nvidia/sycl_cuda_engine.cpp#L145

We should probably release the returned CUcontext there too.

Reply:
This also applies to HIP.

Reply:
I have already changed this
https://github.com/oneapi-src/oneDNN/pull/1765/files#diff-30a9bdfd3a010999776d476eb18b35a4735b6fe2d43e48c06dda79964b6d054cR145
https://github.com/oneapi-src/oneDNN/pull/1765/files#diff-dd4c98e0f1778a87d40952537f1cf07945e28fc30fc7b298b29acbeab7b39592R112

Reply:
I meant this change:
> We should probably release the returned CUcontext there too.

Reply:
Change has been made